Segfault When `air` Command Ran

I got a segfault when I try to run the command `air` but when I reset my PC and run `air` again it refreshes my page but I instantly get segfault
```
watching public
!exclude tmp
watching views
watching views/auth
watching views/components
watching views/css
watching views/home
watching views/layouts
building...
running...
building static files for development
2024/10/19 23:20:29 INFO HTTPS Server Started listenAddr=:3000
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x90 pc=0x5ec12a0]

goroutine 40 [running]:
regexp.(*Regexp).doExecute(0xc0000b2780?, {0x0?, 0x0?}, {0xc00024a120?, 0x5d?, 0x60?}, {0x0?, 0x0?}, 0x0?, 0x0, ...)
        /usr/local/go/src/regexp/exec.go:527 +0x80
regexp.(*Regexp).doMatch(...)
        /usr/local/go/src/regexp/exec.go:514
regexp.(*Regexp).Match(...)
        /usr/local/go/src/regexp/regexp.go:533
github.com/air-verse/air/runner.(*Engine).isExcludeRegex(0xc000222000?, {0xc0000b2780, 0x5d})
        /Users/lidldev/go/pkg/mod/github.com/air-verse/air@v1.61.0/runner/util.go:167 +0xcb
github.com/air-verse/air/runner.(*Engine).watchPath.func1()
        /Users/lidldev/go/pkg/mod/github.com/air-verse/air@v1.61.0/runner/engine.go:256 +0x3ec
created by github.com/air-verse/air/runner.(*Engine).watchPath in goroutine 1
        /Users/lidldev/go/pkg/mod/github.com/air-verse/air@v1.61.0/runner/engine.go:223 +0x23f
```

--- a/runner/config.go
+++ b/runner/config.go
@@ -62,16 +62,6 @@ type cfgBuild struct {
 }
 
 func (c *cfgBuild) RegexCompiled() ([]*regexp.Regexp, error) {
-	if len(c.ExcludeRegex) > 0 && len(c.regexCompiled) == 0 {
-		c.regexCompiled = make([]*regexp.Regexp, 0, len(c.ExcludeRegex))
-		for _, s := range c.ExcludeRegex {
-			re, err := regexp.Compile(s)
-			if err != nil {
-				return nil, err
-			}
-			c.regexCompiled = append(c.regexCompiled, re)
-		}
-	}
 	return c.regexCompiled, nil
 }
 
@@ -317,6 +307,19 @@ func (c *Config) preprocess() error {
 	// Join runtime arguments with the configuration arguments
 	runtimeArgs := flag.Args()
 	c.Build.ArgsBin = append(c.Build.ArgsBin, runtimeArgs...)
+
+	// Compile the exclude regexes if there are any patterns in the config file
+	if len(c.Build.ExcludeRegex) > 0 {
+		regexCompiled := make([]*regexp.Regexp, len(c.Build.ExcludeRegex))
+		for idx, expr := range c.Build.ExcludeRegex {
+			re, err := regexp.Compile(expr)
+			if err != nil {
+				return fmt.Errorf("failed to compile regex %s", expr)
+			}
+			regexCompiled[idx] = re
+		}
+		c.Build.regexCompiled = regexCompiled
+	}
 
 	c.Build.ExcludeDir = ed
 	if len(c.Build.FullBin) > 0 {

